internal/signalfx-agent/pkg/core/dpfilters/overridable.go

```diff
@@ -47,3 +48,26 @@ func (f *overridableDatapointFilter) Matches(dp *datapoint.Datapoint) bool {
 	return (f.metricFilter == nil || f.metricFilter.Matches(dp.Metric)) &&
 		(f.dimFilter == nil || f.dimFilter.Matches(dp.Dimensions))
 }
+
+func (f *overridableDatapointFilter) MatchesMetric(m pmetric.Metric) bool {
+	if f.metricFilter != nil && !f.metricFilter.Matches(m.Name()) {
+		return false
+	}
+	if f.dimFilter != nil {
+		switch m.Type() {
+		case pmetric.MetricTypeGauge:
+			if m.Gauge().DataPoints().Len() == 0 {
+				return false
+			}
+			return f.dimFilter.MatchesMap(m.Gauge().DataPoints().At(0).Attributes())
```

Comment:
I think we should filter out data points from the metric instead of dropping the whole metric if the first datapoint matches the filter. What if the second of third data point matches it instead of the first?

Reply:
Thanks I will change the logic.